doc/source/indexing.rst

```diff
@@ -500,6 +500,62 @@ A list of indexers where any element is out of bounds will raise an
 
 .. _indexing.basics.partial_setting:
 
+Selecting Random Samples
+------------------------
+.. versionadded::0.16.1
+
+A random selection of rows from a Series, DataFrame, or Panel with the ``.sample()`` method. 
+The method accepts a specific number of rows to return, or a fraction of rows. 
+
+.. ipython :: python
+
+   s = Series([0,1,2,3,4,5])
+
+   # When no arguments are passed, returns 5 rows like .head() or .tail()
+   s.sample()
+   
+   # One may specify either a number of rows:
+   s.sample(n = 3)
+   
+   # Or a fraction of the rows:
+   s.sample(frac = 0.5)
+
+By default, ``sample`` will return each row at most once, but one can also sample with replacment
+using the ``replace`` option:
+
+.. ipython :: python
+
+   s = Series([0,1,2,3,4,5])
+		
+   # Without replacement (default):
+   s.sample(n = 6, replace = False)
+	
+   # With replacement:
+   s.sample(n = 6, replace = True)
+
+
+By default, each row has an equal probability of being selected, but if you want rows
+to have different probabilities, you can pass ``sample`` sampling weights. This weights
+can be a list, a numpy array, or a Series, but it must (a) be of the same length as the
+object you are sampling, and (b) the weights must sum to one. For example:
+
+.. ipython :: python
+
+    s = Series([0,1,2,3,4,5])
+    example_weights = [0, 0, 0.2, 0.2, 0.2, 0.4]
+    s.sample(n=3, weights = example_weights)
+	
+
+When applied to a DataFrame, you can use a column of the DataFrame as sampling weights
+by simply passing the name of the column as a string.
+	
+.. ipython :: python
+
+   df = DataFrame({'col1':[9,8,7,6], 'weight_column':[0.5, 0.4, 0.1, 0]})
```

Comment:
You are redefining a `df` here that is used later on (line 610), but that example there was based on how `df` was defined above (line 166). So can you give this here another name (eg `df2`), or repeat the definition of `df` from above below.

You can see the build log of the docs in the travis build (third job): https://travis-ci.org/pydata/pandas/jobs/56058495, scroll down and expand the `$ if [ -f /tmp/doc.log ]; then cat /tmp/doc.log; fi` line